Update CONTRIBUTING.md

mallenexpensify opened the pull request:
LOTS of updates, almost exclusively in "Finding Jobs", I'm going to post in #contributor-management to get bonus eyes on it in case anything isn't _perfect_

<!-- If necessary, assign reviewers that know the area or changes well. Feel free to tag any additional reviewers you see fit. -->

### Details
<!-- Explanation of the change or anything fishy that is going on -->

### Fixed Issues
<!---
Please replace GH_LINK with the link to the GitHub issue this Pull Request is fixing.
Do NOT add the special GH keywords like `fixed` etc, we have our own process of managing the flow.
It MUST be an entire link to the issue; otherwise, the linking will not work as expected.

Make sure this section looks similar to this (you can link multiple issues using the same formatting, just add a new line):

$ https://github.com/Expensify/App/issues/4723
--->
$ GH_LINK

### Tests
<!--- 
Add a numbered list of manual tests you performed that validates your changes work on all platforms, and that there are no regressions present.
Add any additional test steps if test steps are unique to a particular platform.
Manual test steps should be written so that your reviewer can repeat and verify one or more expected outcomes in the development environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image 
--->

### QA Steps
<!--- 
Add a numbered list of manual tests that can be performed by our QA engineers on the staging environment to validate that your changes work on all platforms, and that there are no regressions present.
Add any additional QA steps if test steps are unique to a particular platform.
Manual test steps should be written so that the QA engineer can repeat and verify one or more expected outcomes in the staging environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image 
--->

### Tested On

- [ ] Web
- [ ] Mobile Web
- [ ] Desktop
- [ ] iOS
- [ ] Android

### Screenshots
<!-- Add screenshots for all platforms tested. Pull requests won't be merged unless the screenshots show the app was tested on all platforms.-->

#### Web
<!-- Insert screenshots of your changes on the web platform-->

#### Mobile Web
<!-- Insert screenshots of your changes on the web platform (from a mobile browser)-->

#### Desktop
<!-- Insert screenshots of your changes on the desktop platform-->

#### iOS
<!-- Insert screenshots of your changes on the iOS platform-->

#### Android
<!-- Insert screenshots of your changes on the Android platform-->

mallenexpensify committed:
Update CONTRIBUTING.md

LOTS of updates, almost exclusively in "Finding Jobs", I'm going to post in #contributor-management to get bonus eyes on it in case anything isn't _perfect_

Beamanator commented on CONTRIBUTING.md:
```diff
@@ -27,25 +27,28 @@ If you are hired for an Upwork job and have any job-specific questions, please a
 If you've found a vulnerability, please email security@expensify.com with the subject `Vulnerability Report` instead of creating an issue.
 
 ## Payment for Contributions
-We hire and pay external contributors via Upwork.com. If you'd like to be paid for contributing, please create an Upwork account and apply for a job in the [Upwork issue list](https://www.upwork.com/ab/jobs/search/?q=Expensify%20React%20Native&sort=recency&user_location_match=2). If you think your compensation should be increased for a specific job, you can request a reevaluation by commenting in ...
```
I noticed you deleted the sentence `We hire one contributor for each Upwork job.`, I assume this is because we now may "technically" hire 2 contributors for one job - if one contributor reported the issue and a second contributor fixed the issue, is that correct? If so, I still think it's fine to keep this sentence, or maybe clarify like this: `We hire one contributor to work on each Upwork job.`

mallenexpensify replied:
Yeah, I removed cuz we'll technically be hiring multiple because of the need to pay the proposer.  If you think it's helpful to have `We hire one contributor to work on each Upwork job.` in there, please add it back.

Beamanator replied:
Hmm I think it's find without that sentence actually 👍

Beamanator commented on CONTRIBUTING.md:
```diff
@@ -27,25 +27,28 @@ If you are hired for an Upwork job and have any job-specific questions, please a
 
 #### Finding a job that Expensify posted
-This is the most common scenario for contributors. The Expensify team posts new jobs to the Upwork job list [here](https://www.upwork.com/ab/jobs/search/?q=Expensify%20React%20Native&sort=recency&user_location_match=2). Each job in Upwork has a corresponding GitHub issue, which will include instructions to follow.
+This is the most common scenario for contributors. The Expensify team posts new jobs to the Upwork job list [here](https://www.upwork.com/ab/jobs/search/?q=Expensify%20React%20Native&sort=recency&user_location_match=2). Each job in Upwork has a corresponding GitHub issue, which will include instructions to follow. You can also view [open issues in GitHub with the `Help Wanted` label.](https://git ...
```
```suggestion
This is the most common scenario for contributors. The Expensify team posts new jobs to the Upwork job list [here](https://www.upwork.com/ab/jobs/search/?q=Expensify%20React%20Native&sort=recency&user_location_match=2). Each job in Upwork has a corresponding GitHub issue, which will include instructions to follow. You can also view open jobs by searching for issues in GitHub with the [`Help Wanted` label](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22).
```

mallenexpensify replied:
I'm 👍 for the suggestion, do I Commit suggestion > commit changes if so?  New to this...

NikkiWines replied:
Yep, that's the correct process to go through @mallenexpensify

Beamanator commented on CONTRIBUTING.md:
```diff
@@ -27,25 +27,28 @@ If you are hired for an Upwork job and have any job-specific questions, please a
 
 #### Proposing a job that Expensify hasn’t posted
 
 In this scenario, it’s possible that you found a bug or enhancement that we haven’t posted to the [Upwork job list](https://www.upwork.com/ab/jobs/search/?q=Expensify%20React%20Native&sort=recency&user_location_match=2) or [Github repository](https://github.com/Expensify/App/issues?q=is%3Aissue). This is an opportunity to propose a job, and (optionally) a solution. If it's a valid job proposal th ...
```
How about making this additional edit here?

```suggestion
In this scenario, it’s possible that you found a bug or enhancement that we haven’t posted to the [Upwork job list](https://www.upwork.com/ab/jobs/search/?q=Expensify%20React%20Native&sort=recency&user_location_match=2) or [Github repository](https://github.com/Expensify/App/issues?q=is%3Aissue). This is an opportunity to propose a job, and (optionally) a solution. If it's a valid job proposal that we choose to implement — either internally or via an external contributor — then we will compensate you $250 for identifying and proposing the improvement. Note: If you get assigned the job you proposed you complete the job, this $250 for identifying the improvement is *in addition to* the reward you will be paid for completing the job.
```

mallenexpensify replied:
If you get assigned the job you proposed **and** you complete the job,
Think an 'and' needs to be added

Beamanator replied:
Updated - good catch!

Beamanator replied:
Resolving as this is outdated

Beamanator commented on CONTRIBUTING.md:
```diff
@@ -27,25 +27,28 @@ If you are hired for an Upwork job and have any job-specific questions, please a
-    7. Pause on this step until a member of the Expensify team responds on your issue with next steps.
+1. Check to ensure an issue does not already exist in the [New Expensify Issue list](https://github.com/Expensify/App/issues) or [Upwork job list](https://www.upwork.com/ab/jobs/search/?q=Expensify%20React%20Native&sort=recency&user_location_match=2). Please use your best judgement to search for similar titles and issue descriptions.
+2. If your bug or enhancement matches an existing issue, please feel free to comment on that GitHub issue with your findings if you think it will help solve a problem.
+3. If there is no existing GitHub issue or Upwork job, report the issues in the #expensify-open-source Slack channel, prefix the post with `BUG:` or `Feature Request:`. Include all relevant data, screenshots and examples in the post or thread.
```
Just a suggestion

```suggestion
3. If there is no existing GitHub issue or Upwork job, report the issue(s) in the [#expensify-open-source](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#asking-questions) Slack channel, prefixed with `BUG:` or `Feature Request:`. Include all relevant data, screenshots and examples in the post or thread.
```

mallenexpensify replied:
👍

Beamanator commented on CONTRIBUTING.md:
```diff
@@ -27,25 +27,28 @@ If you are hired for an Upwork job and have any job-specific questions, please a
+2. If your bug or enhancement matches an existing issue, please feel free to comment on that GitHub issue with your findings if you think it will help solve a problem.
+3. If there is no existing GitHub issue or Upwork job, report the issues in the #expensify-open-source Slack channel, prefix the post with `BUG:` or `Feature Request:`. Include all relevant data, screenshots and examples in the post or thread.
+4. After review in #expensify-open-source, if we choose to implement your proposal, a GitHub issue will be created and your Slack handle will be included in the original post after `Issue reported by:`
+5. If an external contributor is hired in Upwork to work on the issue, you will also be hired for the same job. No additional work is needed. If the issue is fixed internally, a dedicated job will be created to hire and pay after the issue is fixed. 
```
```suggestion
5. If an external contributor other than yourself is hired to work on the issue, you will also be hired for the same job in Upwork. No additional work is needed. If the issue is fixed internally, a dedicated job will be created to hire and pay you after the issue is fixed. 
```

mallenexpensify replied:
👍

Beamanator commented on CONTRIBUTING.md:
```diff
@@ -27,25 +27,28 @@ If you are hired for an Upwork job and have any job-specific questions, please a
+3. If there is no existing GitHub issue or Upwork job, report the issues in the #expensify-open-source Slack channel, prefix the post with `BUG:` or `Feature Request:`. Include all relevant data, screenshots and examples in the post or thread.
+4. After review in #expensify-open-source, if we choose to implement your proposal, a GitHub issue will be created and your Slack handle will be included in the original post after `Issue reported by:`
+5. If an external contributor is hired in Upwork to work on the issue, you will also be hired for the same job. No additional work is needed. If the issue is fixed internally, a dedicated job will be created to hire and pay after the issue is fixed. 
+6. Payment will be made 7 days after code is deployed to production if there are no regressions. If a regression occurs, payment will be issued 7 days after all regressions are fixed. 
```
```suggestion
6. Payment will be made 7 days after code is deployed to production if there are no regressions. If a regression is discovered, payment will be issued 7 days after all regressions are fixed. 
```

mallenexpensify replied:
👍

Beamanator reviewed (commented):
Love the changes! Added some comments that could potentially make a few pieces clearer, let me know what you think @mallenexpensify 👍

mallenexpensify commented:
Appreciate the 👀 @Beamanator , I wasn't sure what actions I was supposed to make to approve them so I added a bunch of 👍.

Christinadobrzyn committed:
Update CONTRIBUTING.md

Reworded the "Note: If your solution is also merged, we will give you a $250 bonus for fixing the bug/enhancement." sentence for clarity

Christinadobrzyn committed:
Update CONTRIBUTING.md

Christinadobrzyn commented:
Sorry, I'm not very knowledgeable about pull requests. I don't quite understand how to satisfy this. 

![image](https://user-images.githubusercontent.com/51066321/132309657-0dbe0c9b-a90a-44d3-9441-f9cd08721390.png)

I have read the CLA Document and I hereby sign the CLA

NikkiWines commented:
@Christinadobrzyn no worries! I think you need to make that comment about signing the CLA it's own separate comment though. 

Once you've done that, to re-run the checks you go click on `Details` for the CLI assistant check 
<img width="904" alt="Screen Shot 2021-09-07 at 10 18 18 AM" src="https://user-images.githubusercontent.com/16747822/132310521-b9dd16ba-76bb-458a-a54e-6653c9e5cee1.png">

Then click on `Re-run Jobs` > `Re-run all jobs`
<img width="813" alt="Screen Shot 2021-09-07 at 10 18 27 AM" src="https://user-images.githubusercontent.com/16747822/132310608-32df6519-5e04-49c1-b66a-03d1c9524b56.png">

Beamanator commented:
Yeah @Christinadobrzyn in case it wasn't super clear, first just add a comment with this exact text: `I have read the CLA Document and I hereby sign the CLA`, then follow Nikki's steps - or we can do that for you after you add your comment :D

Christinadobrzyn commented:
I have read the CLA Document and I hereby sign the CLA

Christinadobrzyn commented:
Amazing!! Thanks so much Nikki and Alex! Sorry for being a little slow with this kinda stuff. Done!

Beamanator commented:
Looks great! Once @mallenexpensify commits the requested changes I think we'll be good to go 👍

mallenexpensify committed:
Update CONTRIBUTING.md

Co-authored-by: Alex Beaman <alexbeaman@expensify.com>

mallenexpensify committed:
Update CONTRIBUTING.md

Co-authored-by: Alex Beaman <alexbeaman@expensify.com>

mallenexpensify committed:
Update CONTRIBUTING.md

Co-authored-by: Alex Beaman <alexbeaman@expensify.com>

mallenexpensify committed:
Update CONTRIBUTING.md

Co-authored-by: Alex Beaman <alexbeaman@expensify.com>

mallenexpensify commented:
K, I think I committed all the suggested changes

NikkiWines reviewed (dismissed):
Looks good!

Beamanator commented:
@mallenexpensify think [this comment](https://github.com/Expensify/App/pull/4996#discussion_r702865754) is also useful to add, or did you want to just scratch that one? :D

mallenexpensify commented:
I can't seem to be able to commit it, can you @Beamanator ? 
![image](https://user-images.githubusercontent.com/22508304/132741154-ee8ddcb7-1cdb-4e53-a624-2e63dd3a67ab.png)

Beamanator commented on CONTRIBUTING.md:
```diff
@@ -27,25 +27,28 @@ If you are hired for an Upwork job and have any job-specific questions, please a
 #### Proposing a job that Expensify hasn’t posted
 
-In this scenario, it’s possible that you found a bug or enhancement that we haven’t posted to the [Upwork job list](https://www.upwork.com/ab/jobs/search/?q=Expensify%20React%20Native&sort=recency&user_location_match=2) or [Github repository](https://github.com/Expensify/App/issues?q=is%3Aissue). This is an opportunity to propose a job, and (optionally) a solution. If it's a valid job proposal th ...
+In this scenario, it’s possible that you found a bug or enhancement that we haven’t posted to the [Upwork job list](https://www.upwork.com/ab/jobs/search/?q=Expensify%20React%20Native&sort=recency&user_location_match=2) or [Github repository](https://github.com/Expensify/App/issues?q=is%3Aissue). This is an opportunity to propose a job, and (optionally) a solution. If it's a valid job proposal th ...
```
```suggestion
In this scenario, it’s possible that you found a bug or enhancement that we haven’t posted to the [Upwork job list](https://www.upwork.com/ab/jobs/search/?q=Expensify%20React%20Native&sort=recency&user_location_match=2) or [Github repository](https://github.com/Expensify/App/issues?q=is%3Aissue). This is an opportunity to propose a job, and (optionally) a solution. If it's a valid job proposal that we choose to implement — either internally or via an external contributor — then we will compensate you $250 for identifying and proposing the improvement. Note: If you get assigned the job you proposed **and** you complete the job, this $250 for identifying the improvement is *in addition to* the reward you will be paid for completing the job.
```

Beamanator committed:
Update CONTRIBUTING.md

Beamanator commented:
@mallenexpensify Good call, seems another commit changed that text so my original suggestion couldn't overwrite! I made a new suggestion [here](https://github.com/Expensify/App/pull/4996#discussion_r706049175) and committed it, so we should be good to merge once the tests pass 👍

Beamanator reviewed (approved):
💪 Just waiting for tests to pass :)

Beamanator commented:
Manually cancelled & restarted workflow jobs since the E2E IOS Tests were taking > 3.5 hours (see this: https://github.com/Expensify/App/pull/4996/checks?check_run_id=3565873759)

OSBotify commented:
:hand: This PR was not deployed to staging yet because QA is ongoing. It will be automatically deployed to staging after the next production release.

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/1222462609) to staging by @Beamanator in version: 1.0.96-1 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅ 
🕸 web 🕸|success ✅
